transport/transport-runtime/src/androidTest/java/com/google/android/datatransport/runtime/SchedulerIntegrationTest.java

```diff
@@ -0,0 +1,194 @@
+import com.google.android.datatransport.TransportFactory;
+import com.google.android.datatransport.runtime.backends.BackendRegistry;
+import com.google.android.datatransport.runtime.backends.TransportBackend;
+import com.google.android.datatransport.runtime.scheduling.jobscheduling.SchedulerConfig;
+import com.google.android.datatransport.runtime.scheduling.jobscheduling.Uploader;
+import com.google.android.datatransport.runtime.scheduling.locking.Locker;
+import java.util.UUID;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.stubbing.Answer;
+
+@RunWith(AndroidJUnit4.class)
+public class SchedulerIntegrationTest {
+  private static final String TEST_KEY = "test";
+  private static final String TEST_VALUE = "test-value";
+  private static final String testTransport = "testTransport";
+  private final TransportInternal transportInternalMock = mock(TransportInternal.class);
+  private final TransportBackend mockBackend = mock(TransportBackend.class);
+  private final TransportBackend mockBackend2 = mock(TransportBackend.class);
+  private final BackendRegistry mockRegistry = mock(BackendRegistry.class);
+  private final Context context = InstrumentationRegistry.getInstrumentation().getContext();
+  private final Uploader mockUploader = mock(Uploader.class);
+  private final Locker<Boolean> locker = new Locker<>();
+
+  @Rule
+  public final TransportRuntimeRule runtimeRule =
+      new TransportRuntimeRule(
+          DaggerTestRuntimeComponent.builder()
+              .setApplicationContext(context)
+              .setUploader(mockUploader)
+              .setBackendRegistry(mockRegistry)
+              .setSchedulerConfig(
+                  SchedulerConfig.builder()
+                      .setDelta(500)
+                      .setMaxAllowedTime(100000)
+                      .setMaximumDelay(500)
+                      .build())
+              .setEventClock(() -> 3)
+              .setUptimeClock(() -> 1)
+              .build());
+
+  @Before
+  public void setUp() {
+    when(mockBackend.decorate(any()))
+        .thenAnswer(
+            (Answer<EventInternal>)
+                invocation ->
+                    invocation
+                        .<EventInternal>getArgument(0)
+                        .toBuilder()
+                        .addMetadata(TEST_KEY, TEST_VALUE)
+                        .build());
+    when(mockBackend2.decorate(any()))
+        .thenAnswer(
+            (Answer<EventInternal>)
+                invocation ->
+                    invocation
+                        .<EventInternal>getArgument(0)
+                        .toBuilder()
+                        .addMetadata(TEST_KEY, TEST_VALUE)
+                        .build());
+    doAnswer(
+            (Answer<Void>)
+                i -> {
+                  locker.setResult(true);
+                  i.<Runnable>getArgument(2).run();
```

Comment:
Please document the above two lines:
* what the locker is needed for and how it synchronizes the tests with the service/broadcast receiver
* why we need to the the passed-in runnable.